packages/workbox-core/src/_private/allSettled.ts

```diff
@@ -0,0 +1,46 @@
+import '../_version.js';
+
+export interface PromiseResolution<T> {
+  status: 'fulfilled';
+  value: T;
+}
+
+export interface PromiseRejection {
+  status: 'rejected';
+  reason: unknown;
+}
+
+export type PromiseResult<T> = PromiseResolution<T> | PromiseRejection;
+
+/**
+ * Promise.allSettled polyfill based on
+ *
+ * https://github.com/es-shims/Promise.allSettled/blob/main/implementation.js
+ *
+ * which is (c) 2019 Jordan Harband and used under the terms of the MIT license.
+ *
+ * @private
+ */
+function allSettled<T>(
```

Comment:
Alternatively to my suggestion below, this could do something like:

```suggestion
const allSettled = Promise.allSettled ? Promise.allSettled.bind( Promise ) : function allSettledImpl<T>(
```